### PR TITLE
Add tests for queue shortcut functions

### PR DESCRIPTION
**Pull request by nolmoonen**
Adds tests for queue shortcut functions according to test plan #407.

Several parts of the tests are compile-time disabled for ComputeC++ and DPCPP to ensure that the tests build successfully.

### CODE REVIEW

File: tests/queue/queue_shortcuts_explicit_fp16.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#include "../common/common.h"
+#include "../common/disabled_for_test_case.h"
+#include "queue_shortcuts_explicit.h"
+
+namespace queue_shortcuts_explicit_fp16 {
+
+using namespace sycl_cts;
+using namespace queue_shortcuts_explict;
+
+// hipSYCL, DPCPP not tested
```
Comment (bader) [inline tests/queue/queue_shortcuts_explicit_fp16.cpp:31]: We have both of them in CI, so I suggest enable and check the status in the logs.

File: tests/queue/queue_shortcuts_explicit_fp16.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+DISABLED_FOR_TEST_CASE(hipSYCL, DPCPP)
+("queue shortcuts explicit copy fp16", "[queue]")({
+  auto queue = util::get_cts_object::queue();
+  using avaliability =
+      util::extensions::availability<util::extensions::tag::fp16>;
+  if (!avaliability::check(queue)) {
+    WARN("Device does not support half precision floating point operations");
```
Comment (bader) [inline tests/queue/queue_shortcuts_explicit_fp16.cpp:38]: We usually warn that test case is skipped:
```c++
    WARN(
        "Device does not support half precision floating point operations. "
        "Skipping the test case.");
```

File: tests/queue/queue_shortcuts_usm.h [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+                element_count * sizeof(T));
+
+    // set initial values of device-allocated usm buffer without using shortcuts
+    T* d_ptr = sycl::malloc_device<T>(element_count, queue);
+    queue.submit([&](sycl::handler& cgh) {
+      cgh.memcpy(d_ptr, h_ptr.get(), element_count * sizeof(T));
+    });
+    queue.wait();
+
+    // perform the memset from d_src to d_dest using shortcuts
```
Comment (bader) [inline tests/queue/queue_shortcuts_usm.h:95]: There are no `d_src` and `d_dest` variables in this function.

File: tests/queue/queue_shortcuts_usm.h [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+    sycl::event event = helper(d_ptr, test);
+    event.wait();
+
+    // copy destination buffer back to host without shortcut functions
```
Comment (bader) [inline tests/queue/queue_shortcuts_usm.h:99]:
```suggestion
    // copy buffer back to host without shortcut functions
```

File: tests/queue/queue_shortcuts_usm.h [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+    std::fill(h_ptr.get(), h_ptr.get() + element_count, t_init);
+
+    // set initial values of device-allocated usm buffer without using shortcuts
+    T* d_ptr = sycl::malloc_device<T>(element_count, queue);
+    queue.submit([&](sycl::handler& cgh) {
+      cgh.memcpy(d_ptr, h_ptr.get(), element_count * sizeof(T));
+    });
+    queue.wait();
+
+    // perform the memset from d_src to d_dest using shortcuts
```
Comment (bader) [inline tests/queue/queue_shortcuts_usm.h:132]: Please, update comments in this function. They seem to be copied from `run_memset` function.

**Review by bader — CHANGES_REQUESTED** [review]
Please, fix compilation with ComputeCPP.

File: tests/queue/queue_shortcuts_explicit_fp16.cpp (hunk 1 above)
Comment (nolmoonen) [reply tests/queue/queue_shortcuts_explicit_fp16.cpp]: I have removed the checks for hipSYCL, but the queue tests (already) are in the CI filter file for hipSYCL, so the test will not be executed.

I have made the comment for DPCPP more specific, the check is in place due to the test not compiling.

File: tests/queue/queue_shortcuts_explicit_fp16.cpp (hunk 2 above)
Comment (nolmoonen) [reply tests/queue/queue_shortcuts_explicit_fp16.cpp]: I updated the warnings, and added warnings in places where `SYCL_CTS_COMPILING_WITH_XXX` macros are used.

File: tests/queue/queue_shortcuts_usm.h (hunk 3 above)
Comment (nolmoonen) [reply tests/queue/queue_shortcuts_usm.h]: I updated the comments.

File: tests/queue/queue_shortcuts_usm.h (hunk 4 above)
Comment (nolmoonen) [reply tests/queue/queue_shortcuts_usm.h]: I updated the comments.

File: tests/queue/queue_shortcuts_usm.h (hunk 5 above)
Comment (nolmoonen) [reply tests/queue/queue_shortcuts_usm.h]: I updated the comments.

**Comment by nolmoonen** [comment]
> Please, fix compilation with ComputeCPP.

I fixed the error locally, but it looks like I need someone to start the CI manually to verify this.

**Review by bader — APPROVED** [review]

**Review by ProGTX — APPROVED** [review]

**Pull request merged**